web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php

```diff
@@ -136,36 +161,65 @@ public function consent(Request $request): array {
     ];
   }
 
-  public function login(Request $request): array {
+  /**
+   * Login through OIDC if consent is given.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *
+   * @return \Drupal\Core\Routing\TrustedRedirectResponse|\Symfony\Component\HttpFoundation\RedirectResponse
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function login(Request $request): TrustedRedirectResponse|RedirectResponse {
+    // Get login URL.
     $client_name = 'connection';
     $this->session->saveOp('login');
     $client = $this->entityTypeManager->getStorage('openid_connect_client')->loadByProperties(['id' => $client_name])[$client_name];
     $plugin = $client->getPlugin();
     $scopes = 'openid profile';
     $response = $plugin->authorize($scopes);
-
     $url = $response->getTargetUrl();
+
+    // Go to login or front page if no consent.
+    if (isset($_REQUEST['consent']) && $_REQUEST['consent'] == 'consent_given' && isset($url)) {
```

Comment:
Use `$request` to get query string parameters:

```suggestion
    if ('consent_given' === $request->get('consent') && isset($url)) {
```

It may also be worth adding `const`s for `'consent'` and `'consent_given'`:

```
    if (self::CONSENT_VALUE === $request->get(self::CONSENT_NAME) && isset($url)) {
```